main.tf

```diff
@@ -26,12 +25,11 @@ resource "aws_s3_bucket" "this" {
 }
 
 resource "aws_s3_bucket_public_access_block" "this" {
-  count = var.create_bucket ? 1 : 0
 
-  bucket = aws_s3_bucket.this[0].id
+  bucket = aws_s3_bucket.this.id
 
   block_public_acls       = var.public_access_block.block_public_acls
   block_public_policy     = var.public_access_block.block_public_policy
   ignore_public_acls      = var.public_access_block.ignore_public_acls
   restrict_public_buckets = var.public_access_block.restrict_public_buckets
-}
+}
```

Comment:
the CI tests are failing because files are missing a final newline. recommend installing the editorconfig plugin for your editor.